java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/utils/ArrowFlightConnectionConfigImpl.java

```diff
@@ -38,6 +38,9 @@
  * A {@link ConnectionConfig} for the {@link ArrowFlightConnection}.
  */
 public final class ArrowFlightConnectionConfigImpl extends ConnectionConfigImpl {
+  private static final ArrowFlightConnectionProperty[] ARROW_FLIGHT_CONNECTION_PROPERTY_VALUES
```

Comment:
(nit) Maybe we can use a truly immutable type like `EnumSet` + `Collections#unmodifiableSet()` instead of an array.